command/main.go

```diff
@@ -264,3 +254,47 @@ func printCommand(w io.Writer, name string, cmdFn cli.CommandFactory) {
 	}
 	fmt.Fprintf(w, "    %s\t%s\n", name, cmd.Synopsis())
 }
+
+func generateCurlString(exitCode int, runOpts *RunOptions) int {
+	if exitCode == 0 {
+		fmt.Fprint(runOpts.Stderr, "Could not generate cURL command")
+		return 1
+	} else {
+		if api.LastOutputStringError == nil {
+			if exitCode == 127 {
+				// Usage, just pass it through
+				return exitCode
+			}
+			fmt.Fprint(runOpts.Stderr, "cURL command not set by API operation; run without -output-curl-string to see the generated error\n")
+			return exitCode
+		}
+		if api.LastOutputStringError.Error() != api.ErrOutputStringRequest {
+			runOpts.Stdout.Write([]byte(fmt.Sprintf("Error creating request string: %s\n", api.LastOutputStringError.Error())))
+			return 1
+		}
+		runOpts.Stdout.Write([]byte(fmt.Sprintf("%s\n", api.LastOutputStringError.CurlString())))
+		return 0
+	}
+}
+
+func generatePolicy(exitCode int, runOpts *RunOptions) int {
+	if exitCode == 0 {
+		fmt.Fprint(runOpts.Stderr, "Could not generate policy")
+		return 1
+	} else {
+		if api.LastOutputPolicyError == nil {
+			if exitCode == 127 {
+				// Usage, just pass it through
+				return exitCode
+			}
+			fmt.Fprint(runOpts.Stderr, "Required policy not clear from API operation\n")
```

Comment:
It does read strangely you're right, I'll change the first part to what you said. 

However, the reason I removed the "run without -output-policy to see the generated error" message I had in there before is because I realized the user often won't be able to run the command without the flag, they'll just get a 403, which may be a red herring in terms of what happened. (Since I imagine people will often be running this flag to see _what_ privileges their token needs in order to get past the 403.)

Reply:
That seems fair. Hopefully we can do better than re-running anyway if the comment about saving stderr works out.